### PR TITLE
Add HPT-DINO action-expert algorithm

### PR DESCRIPTION
**Pull request by ElmoPA**
Two-stage policy:
- Stage 1 (world model with trunk): BOS DINO + sampled language prompt -> EOS
  DINO via shared FMPolicy flow head. With patch-token DINO embeddings
  (N=196 for ViT-B/16 at 224x224), action_horizon and act_seq are 196 and
  act_dim is the DINO hidden dim (768).
- Stage 2 (action expert, no trunk): GT EOS DINO + obs_t (front + wrist + 14d
  wrist-frame YPR proprio) -> 100-step action chunk via shared FMPolicy.
Training is staged, not joint: stage1_steps then stage2_steps, with each
stage's pretrained weights loadable independently.

Adds:
- HPTDinoAE algo + ActionExpertModel (egomimic/algo/hpt_dino_ae.py).
  Mode-aware annotation sampling: random during training,
  eval_annotation_index during validation. Robust pretrained loading
  tolerant to state_dict / nets.policy. prefixes.
- Qwen3TextEncoder stem with freeze_backbone toggle (egomimic/models/hpt_nets.py).
- Eva/Aria dino_action_expert keymaps + get_dino_ae_transform_list.
  Keymaps respect norm_mode=True (drops camera + annotation keys for the
  norm-stats DataLoader, which uses default_collate and would otherwise
  crash on variable-length annotation lists).
- Action-expert dataset hardening: iterative resample (no recursion),
  last-frame action padding (avoids zero-norm quaternions in scipy),
  EOS clamp to total_frames-1 (handles Python-style exclusive end_idx),
  TYPE_CHECKING import to break the multi <-> action_expert circular cycle.
- Data schematic + data + model (shared head) + top-level train + viz
  configs for the two-stage pipeline. eva_aria_dino_ae.yaml inherits from
  cotrain_pi_base and overrides only the resolver, key_map, transform_list
  and filters.

Co-Authored-By: Claude Opus 4.7 <noreply@anthropic.com>

### CODE REVIEW

**Comment by ElmoPA** [comment]
* **#378** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/378?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#377** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/377?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#376** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/376?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/376?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#372** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/372?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#390](https://github.com/GaTech-RL2/EgoVerse/pull/390) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/390?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#371** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/371?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#375** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/375?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
